CONTRIBUTING.md

```diff
@@ -0,0 +1,13 @@
+# Style guide
+## Enforcement
+The project uses [CheckStyle](https://checkstyle.sourceforge.io/) to enforce style guidelines. We use the standard [Google Checks](https://checkstyle.sourceforge.io/google_style.html) with some additional custom rules defined within [the project](./etc/checkstyle-custom_checks.xml).   
+
+If we can't automatically enforce a formatting convention then we don't want to argue about it on pull requests :D 
+
+## Validating invariants
+### Public API
+To enforce invariants in the public API, explicitly test invariant and throw `IllegalArgumentException` or more specific `RuntimeException` as appropriate. 
+
+### Internal Code
+Asserts can be used for documenting invariants in internal code, private methods and the s. As the test suite runs with assertions enabled, this can help to spot bugs which would otherwise go unnoticed.
```

Comment:
```suggestion
`assert` can be used for documenting invariants in internal code, private methods and the s. As the test suite runs with assertions enabled, this can help to spot bugs which would otherwise go unnoticed.
```

I'm not sure that "and the s" is supposed to be.